business/openid_auth.go

```diff
@@ -783,10 +787,30 @@ func createHttpClient(toUrl string) (*http.Client, error) {
 		return nil, err
 	}
 
+	// Check if there is a user-configured custom certificate for the OpenID Server. Read it, if it exists
+	var cafile []byte
+	if _, customCaErr := os.Stat(OpenIdServerCAFile); customCaErr == nil {
+		var caReadErr error
+		if cafile, caReadErr = ioutil.ReadFile(OpenIdServerCAFile); caReadErr != nil {
+			return nil, fmt.Errorf("failed to read the OpenId CA certificate: %w", caReadErr)
+		}
+	} else if !errors.Is(customCaErr, os.ErrNotExist) {
+		log.Warningf("Unable to read the provided OpenID Server CA file (%s). Ignoring...", customCaErr.Error())
+	}
+
 	httpTransport := &http.Transport{}
-	if cfg.InsecureSkipVerifyTLS {
+	if cfg.InsecureSkipVerifyTLS || cafile != nil {
+		var certPool *x509.CertPool
+		if cafile != nil {
+			certPool = x509.NewCertPool()
+			if ok := certPool.AppendCertsFromPEM(cafile); !ok {
+				return nil, fmt.Errorf("supplied OpenId CA file cannot be parsed")
+			}
+		}
+
 		httpTransport.TLSClientConfig = &tls.Config{
-			InsecureSkipVerify: true,
+			InsecureSkipVerify: cfg.InsecureSkipVerifyTLS,
+			RootCAs:            certPool,
```

Comment:
In the if-stmt on 802, I see we get in this if-clause if "InsecureSkipVerifyTLS" is true **even if** cafile == nil.

If cafile is nil, that means certPool is never set (thus is "nil" pointer). When you go to set this RootCAs field, it is getting set to a nil point.

Is that an error? That seems like it could be a problem. Have you validated that code path? i.e. set InsecureSkipVerifyTLS to true and set cafile to nil

Reply:
It is similar to what the HTTP util is doing.
See: https://github.com/kiali/kiali/blob/3e032198c053f4cc9565c7a8f0421f51561b31f9/util/httputil/http_util.go#L103-L124

So, I think it shouldn't fail. But I suspect it may have an unwanted behavior. Let me double check.

Reply:
It does not fail with the nil value.